testing/snapshot.ts

```diff
@@ -33,51 +72,240 @@ function serialize(actual: unknown): string {
 }
 
 /**
- * Write updates to the snapshot file.
+ * Converts a string to a valid JavaScript string which can be wrapped in backticks.
+ *
+ * @example
  *
- * @param context Assert snapshot context
+ * "special characters (\ ` $) will be escaped" -> "special characters (\\ \` \$) will be escaped"
+ */
+function escapeStringForJs(str: string) {
+  return str
+    .replace(/\\/g, "\\\\")
+    .replace(/`/g, "\\`")
+    .replace(/\$/g, "\\$");
+}
+
+let _mode: SnapshotMode;
+/**
+ * Get the snapshot mode.
  */
-function writeSnapshotFileSync(context: AssertSnapshotContext) {
-  const buf = ["export const snapshot = {};"];
-  function escapeStringForJs(str: string) {
-    return str
-      .replace(/\\/g, "\\\\")
-      .replace(/`/g, "\\`")
-      .replace(/\$/g, "\\$");
+function getMode(options: SnapshotOptions) {
+  if (options.mode) {
+    return options.mode;
+  } else if (_mode) {
+    return _mode;
```

Comment:
Can you add a test that will exercise this line?

Reply:
This wasn't covered because I made a mistake in the implementation. This is fixed now and this line is covered without the need for a specific test.